src/service/resources-impl.js

```diff
@@ -657,8 +712,11 @@ export class ResourcesImpl {
     if (resource.isBuilt()) {
       resource.pauseOnRemove();
     }
-    if (this.intersectionObserver_) {
-      this.intersectionObserver_.unobserve(resource.element);
+    if (this.oneViewportObserver_) {
+      this.oneViewportObserver_.unobserve(resource.element);
```

Comment:
Are there times where we call `removeResource` without having the element also removed from the DOM? If so, why do we need to manually call `unobserve`, I'd expect IntersectionObserver to handle a removed element gracefully.

Reply:
Check out the callers of `Resources.remove()`: https://github.com/ampproject/amphtml/blob/6b439aec75c1d9b52ba19435f6730b27a457bf42/src/custom-element.js#L949-L950